packages/web-react/src/types/grid.ts

```diff
@@ -0,0 +1,22 @@
+import { ElementType, JSXElementConstructor } from 'react';
+import { ChildrenProps, StyleProps } from './shared';
+
+export type GridColumns = 1 | 2 | 3 | 4 | 6 | 12;
+
+export interface GridElementTypeProps<T extends ElementType = 'div'> {
+  /**
+   * The HTML element or React element used to render the Grid, e.g. 'div'.
+   * @default 'div'
+   */
+  elementType?: T | JSXElementConstructor<any>;
+}
+
+export interface GridProps<T extends ElementType = 'div'> extends GridElementTypeProps<T> {}
+
+export interface SpiritGridProps<T extends ElementType = 'div'> extends GridProps<T>, ChildrenProps, StyleProps {
+  mobile?: GridColumns;
+  tablet?: GridColumns;
+  desktop?: GridColumns;
+  cols?: GridColumns;
+  narrow?: boolean;
```

Comment:
We expect the number of predefined layouts to grow in the near future, at least we can be quite sure there will be a sidebar layout or two. Therefore I'd suggest to change this boolean prop to an enum like `layout?: GridLayouts` with `GridLayouts` offering just the `narrow` option for now. This way it's not going to be a BC once we add more layouts.

Furthermore, the predefined layouts are **mutually exclusive** with other column settings: when `narrow` layout is applied, you cannot set the number of columns with other props because the layouts always expect 12 column grid. (The CSS is now broken when you do that, we are on it.) Do you think it should be covered in the API? Or in the docs? Or it doesn't matter once we fix the CSS?

Reply:
Thank you for your advice. I've changed it to `layout` prop. But, mutually exclusive props are a little bit a nightmare in typescript. It can be done using `type` but not `interface` (https://github.com/microsoft/TypeScript/issues/14094), as far I know.

All I can do now is to raise an error with message of incompatibility.

Reply:
> raise an error with message of incompatibility

Nice! Maybe we could just raise a warning in such case?

Reply:
Fixed.